x/evm/keeper/state_transition.go

```diff
@@ -310,15 +312,16 @@ func (k *Keeper) ApplyMessageWithConfig(
 		stateDB.PrepareAccessList(msg.From(), msg.To(), evm.ActivePrecompiles(rules), msg.AccessList())
 	}
 
+	scaledMsgValue := utils.Convert18To6Decimals(*msg.Value())
```

Comment:
within the EVM we should always have 18 decimals. I believe we don't need this conversion here no? cause we're managing the conversion from 6 to 18 on the getBalance, and from 18 to 6 in the setBalance funcs

Reply:
This comes up always as 18 decimals since that's the default `msg.Value` decimals. It doesn't work without it send transactions are out of balance errors since this is being propagated down the call stack ?